### ISSUE TITLE
Aggregate issue: builds stuck in "pending" state

### ISSUE THREAD

**Issue by vito** [concourse/concourse#844]
There have been a few reports lately of jobs being unable to schedule. It's been difficult for us to progress on this as we never see it ourselves and there's generally not enough information provided to find a smoking gun. Which is no one's fault really - we haven't given great instructions as to what information would be valuable and how to collect it! So, that's what this issue is for.

If you're seeing this problem, please include a screenshot of the build preparation section of the build view.

Also answer the following:

* Are all of your workers present? (`fly workers`)
* Is there a `check` container present for each of the inputs to your job? (`fly hijack -c pipeline-name/resource-name`)
  * Do any of the `check` containers have a running `/opt/resource/check` process? If so, that may be hanging. What resource type is it?
* What is the uptime of your workers and ATC?
* Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?
* Which IaaS?
  * If you're on GCP, have you configured the MTU of your workers to be `1460` to match the VM? If not, it defaults to `1500`, which would cause things to hang.
  * Can you reach the workers from your ATC? (`curl http://<worker ip:port>/containers`) You can collect the IP + port from `fly workers -d`.

In addition to that, the **most valuable information will be stack dumps of both the ATC and the TSA**.

You can collect a stack dump from the ATC by running:

```sh
curl http://127.0.0.1:8079/debug/pprof/goroutine?debug=2
```

...and from the TSA by sending `SIGQUIT` and collecting the output from `stderr`. Note that if you're running the binaries, the above `curl` command will include the TSA's stack, so don't worry about getting it separately. Also note that `SIGQUIT` will kill the TSA process, so you'll need to bring it back after. (While you're at it, let us know if that fixed it. :P)

Thanks all for your patience, sorry we're taking so long to get to the bottom of this.

**Comment by freelock** [comment]
Oh, excellent, I'll be sure to collect the requested info next time we hit this. Which will probably be in about 2 weeks, when we run the next round of updates through our testing pipeline.

**Comment by databus23** [comment]
For us this is definitely triggered by #750.  Whenever we loose a worker with a deadlocked SSH connection we usually have 2-3 jobs hanging in pending with everything checked like [this](https://github.com/concourse/concourse/issues/785#issuecomment-265145906) or [this](https://github.com/concourse/concourse/issues/785#issuecomment-265148208).

Each time at least one check container of the jobs inputs is missing. The only way to bring back the missing containers seems to be restarting the `web` component.

My assumption is that the check containers that are missing where located on the missing worker with the deadlocked SSH connection on the TSA side. While resolving #750 would likely fix this issue for us I think concourse should be able to recover from this even without it. The worker with the stuck connection is long gone and so is the check container. The scheduler should in priciple be able reschedule the check container on a different worker.

**Comment by vito** [comment]
@databus23 Agreed that a lot of this may go back to #750. Thanks for the additional details.

Architecturally, Concourse should handle this just fine. It'll see that the worker's gone and that there's no `check` container, and the ATC will just make a new one on the remaining workers. Unfortunately, what's happening here is that a deadlock in the TSA turns into a deadlock in the ATC, as the connection is forwarded through the TSA. Worse still, this happens with a database-level lock held (so we don't run `check` redundantly), so another ATC can't even make its own attempt. This is why things grind to a halt.

Unfortunately this kind of thing is tricky to write defensive code against. It's technically a "logic error" deadlock over a healthy connection, so even things like TCP keepalives won't help. Restarting the ATC should fix it, which is not ideal, but at least shows that the Concourse cluster as a whole is not permanently broken due to a worker leaving.

We could start configuring request timeouts but I'm generally wary of those as they'll often either mask real issues, leaving them unfixed (like this one), or become problematic for folks that just genuinely have a slow network or slow worker hardware. We could have a large default or make it configurable, but then the issue would be "pending builds are sometimes slow to start", which is even *harder* to trace back to a smoking gun.

**Comment by vrecan** [comment]
why not use https://godoc.org/golang.org/x/net/context to deal with this using either deadlines or timeouts?

**Comment by sesmith177** [comment]
We are frequently running into this as well.

All our workers are here:

```
fly -t buildpacks workers
name                                  containers  platform  tags  team  state
45a00bc0-0e90-489c-83a8-4a70c49f8329  114         linux     none  none  running
6afcf860-86de-458c-ad33-6d9656914b83  116         linux     none  none  running
ab42c661-826c-4ca1-b1fb-d17842efdc26  126         linux     none  none  running
dce38b0a-a3e1-45bc-828f-c5e765bb42e3  129         linux     none  none  running
e20a836f-17b8-4b37-8b6b-2243219f7445  88          linux     none  none  running
fee325ad-4186-46ab-ac4f-b10bf65a53d0  121         linux     none  none  running
```

There is no check container for the nightly-trigger input to the hanging [job](https://buildpacks.ci.cf-app.com/teams/main/pipelines/edge-1/jobs/remove-environment/builds/47):

```
$ fly -t buildpacks intercept -c edge-1/resource-pools
bash-4.3# exit
exit
fly -t buildpacks intercept -c edge-1/nightly-trigger
no containers matched your search parameters!

they may have expired if your build hasn't recently finished.
```

An ATC stack dump is [here](https://gist.github.com/cf-buildpacks-eng/6ea365d11722f1647166d0b5d02b7180).

This Concourse installation is running on GCP with the MTU correctly configured

Rebooting the web vm usually resolves this for us.

**Comment by vito** [comment]
@sesmith177 That's an interesting one. It looks like a bunch of hanging connections to Postgres, not Garden. Any more information you can collect knowing that? Maybe in your Postgres logs or VM info? Check its uptime? Maybe the VM faulted and BOSH recreated it but by then the connections were already hanging, for example. I don't recall if we have TCP keepalives for postgres or generally any way to detect a dead connection.

**Comment by sesmith177** [comment]
@vito we use an external RDS postgres instance, not a VM managed by BOSH.

Looking in the logs from this morning we see:

```
2017-01-09 14:33:52 UTC:175.174.196.104.bc.googleusercontent.com(51690):atc@atc:[12294]:LOG: could not send data to client: Connection timed out
2017-01-09 14:33:52 UTC:175.174.196.104.bc.googleusercontent.com(51690):atc@atc:[12294]:STATEMENT: 
SELECT p.id, p.name, p.config, p.version, p.paused, p.team_id, p.public, t.name as team_name
FROM pipelines p
INNER JOIN teams t ON t.id = p.team_id
WHERE p.id = $1

2017-01-09 14:33:52 UTC:175.174.196.104.bc.googleusercontent.com(51690):atc@atc:[12294]:FATAL: connection to client lost
2017-01-09 14:33:52 UTC:175.174.196.104.bc.googleusercontent.com(51690):atc@atc:[12294]:STATEMENT: 
SELECT p.id, p.name, p.config, p.version, p.paused, p.team_id, p.public, t.name as team_name
FROM pipelines p
INNER JOIN teams t ON t.id = p.team_id
WHERE p.id = $1
```

**Comment by mkb** [comment]
- Are all of your workers present? (fly workers)

Yes

- Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)

Yes

- Do any of the check containers have a running /opt/resource/check process? If so, that may be hanging. What resource type is it?

No

- What is the uptime of your workers and ATC?

Just shy of two hours.

-Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?

Running in the Vagrant image so I assume the latter.

- Which IaaS?

Vagrant. I have also observed the same behavior with BOSH/vSphere.

- Can you reach the workers from your ATC? (curl http://<worker ip:port>/containers) You can collect the IP + port from fly workers -d.

Yes.

-------

Stack dumps enclosed.

[concourse-stack-dumps.txt](https://github.com/concourse/concourse/files/694702/concourse-stack-dumps.txt)

If I can provide any additional info, just shout. Thanks!

**Comment by DennisDenuto** [comment]
- Are all of your workers present? (fly workers)

yes

- Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)

no

- What is the uptime of your workers and ATC?

```
atc> uptime
 21:58:35 up 14:56,  1 user,  load average: 0.11, 0.15, 0.21
worker> uptime
 21:59:12 up 14:57,  1 user,  load average: 0.02, 0.02, 0.00
```

- Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?

both


- Which IaaS?

AWS + vsphere



- Can you reach the workers from your ATC? (curl http://<worker ip:port>/containers) You can collect the IP + port from fly workers -d.

Yes

See atc stack [dump.txt](https://github.com/concourse/concourse/files/694778/dump.txt)

**Comment by mkb** [comment]
I should clarify that in my case the hang is consistently when checking Git resources, gists in particular.

**Comment by davewalter** [comment]
- Screenshot
![screen shot 2017-01-09 at 3 32 44 pm](https://cloud.githubusercontent.com/assets/33302/21787969/f13f9180-d680-11e6-98d8-3e4fabbfbdc5.png)


- Are all of your workers present? (fly workers)
Yes

- Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)
No

- What is the uptime of your workers and ATC?
20 days for the ATC. Between 6 and 20 days for the works.

- Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?
Through the TSA

- Which IaaS?
AWS

- Can you reach the workers from your ATC? (curl http:///containers) You can collect the IP + port from fly workers -d.
Yes

- Debug output
[atc-dump.txt](https://github.com/concourse/concourse/files/694890/atc-dump.txt)

**Comment by vito** [comment]
@DennisDenuto @davewalter @sesmith177 Could you include the TSA stack dump as well if it's a separate process?

**Comment by freelock** [comment]
> If you're seeing this problem, please include a screenshot of the build preparation section of the build view.
![selection_558](https://cloud.githubusercontent.com/assets/46937/21830825/5d6d608e-d754-11e6-924b-3d9f2197d045.png)


> Also answer the following:

> Are all of your workers present? (fly workers)
```
name         containers  platform  tags  team  state  
Freelock-06  25          linux     lan   none  running


the following workers have not checked in recently:

name                   containers  platform  tags  team  state  
worker07.freelock.com  55          linux     none  none  stalled

```
(worker07 is our check worker, all jobs and resources run on workers tagged with "lan")

> Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)

Expired (or unreachable due to check worker not available).

> Do any of the check containers have a running /opt/resource/check process? If so, that may be hanging. What resource type is it?

N/A

> What is the uptime of your workers and ATC?

ATC: 2 days 6:59 (and concourse web was restarted less than a day ago)
Workers: 4 days 9:54

> Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?

Binary default.

> Which IaaS?

ATC and untagged worker on GCE, same vlan. Tagged worker on our LAN.

> If you're on GCP, have you configured the MTU of your workers to be 1460 to match the VM? If not, it defaults to 1500, which would cause things to hang.

??? is GCP == Google Cloud Engine?

It looks like this is indeed a problem, I'm seeing MTU of 1460 on the main interface and 1500 on all the workers. 

I'm resetting this, restarting everything, and will see whether that makes a difference.

**Comment by devdavidkarlsson** [comment]
I got this when I had a resource that was not working (missing opt folder) in one of my pipelines, it seemed to block the entire ATC.

**Comment by pdelagrave** [comment]
We frequently have this issue. Killing the job and restarting it won't fix it.
We used the `set_mtu` job of the `networking` bosh release yesterday to set 1460 as the MTU for all our workers and we still see the problem.

We can give access to our concourse bosh deployment where there's currently jobs stuck in that state. Contact me on slack for the credentials.

**Comment by jraqula** [comment]
i experienced this issue with one of my ~10 workers. It began when I recreated a worker due to unresponsive EBS drive. However, it was solved by restarting the ATC node.

**Comment by rhenning** [comment]
Hello, Concourse folks! We've been seeing this one with increasing regularity since upgrading to the most recent release (but we came from 1.6 so :man_shrugging:). Anyway, here are the requested diagnostics. Please let us know if we can provide any more information or help out. Thanks!

> Screenshot of a hanging build

![screen shot 2017-02-10 at 5 02 39 pm](https://cloud.githubusercontent.com/assets/1392051/22845915/f3d95ca0-efb2-11e6-83b7-16603770b81e.png)

> ATC stack trace

[atc_stack.txt](https://github.com/concourse/concourse/files/767906/atc_stack.txt)

> Are all of your workers present? (fly workers)

Yes

```
 fly -t aae-concourse workers -d
name                              containers  platform  tags  team  state    garden address   baggageclaim url        resource types                                                                                                                                        
concourse-ui-ndc-as-b-blue        75          linux     none  none  running  127.0.0.1:53457  http://127.0.0.1:42420  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
concourse-worker-ndc-as-b-blue-0  63          linux     none  none  running  127.0.0.1:50318  http://127.0.0.1:51040  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
concourse-worker-ndc-as-b-blue-1  79          linux     none  none  running  127.0.0.1:51982  http://127.0.0.1:39904  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
```

> Is there a check container present for each of the inputs to your job?

No check containers. All containers for this job appear to be for builds other than the hanging one.

> Uptime of workers and ATC

ATC:
Process: 2.5h, System: 7d, OS: Ubuntu 14.04, Kernel: 3.19.0-79-generic

Workers:
concourse-ui-ndc-as-b-blue: Process: 7h, System: 7d, OS: Ubuntu 14.04, Kernel: 3.19.0-79-generic
concourse-worker-ndc-as-b-blue-0: Process: ~21d, System: 28d, OS: Ubuntu 14.04, Kernel: 3.19.0-79-generic
concourse-worker-ndc-as-b-blue-1: Process: ~21d, System: 28d, OS: Ubuntu 14.04, Kernel: 3.19.0-79-generic

> Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?

Forwarded thru the ATC (binary default)

> IaaS

OpenStack

> Can you reach the workers from your ATC?

Yes

**Comment by rhenning** [comment]
Fascinating... I went home and hopped on the VPN only to discover that our "hanging" `pending` builds, one of which is screencap'd above, have updated, about 3h after being triggered.  Here's the view now:

<img width="541" alt="screen shot 2017-02-10 at 7 10 40 pm" src="https://cloud.githubusercontent.com/assets/1392051/22848813/24d6f30a-efc5-11e6-9f9b-d62ad2f2848a.png">

**Comment by vito** [comment]
FYI: 2.7 has shipped recently which should fix #750, a likely candidate for the root cause of this. For those affected by this issue, please upgrade and let us know if it's fixed.

**Comment by ericroy** [comment]
I'm seeing this in 2.7 consistently when using a custom resource

**Comment by vito** [comment]
@ericroy Could you run through the diagnostics in the description? Sounds like it could be something different.

**Comment by ericroy** [comment]
Sure, I will do it when I get back to my computer in an hour or so

**Comment by ericroy** [comment]
Ok, so I'm using the vagrant image that you provide.  I deleted my pipeline, restarted the vagrant box, and re-registered the pipeline.  Then, I kickoff my job and wait a few minutes.  It hangs in the 'preparing' phase.  Here are the requested details:

- Are all of your workers present? (fly workers)

Yes:
```
$ ./fly -t dingomini workers
name       containers  platform  tags  team  state
concourse  0           linux     none  none  running
```

- Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)

No:
```
$ ./fly -t dingomini hijack -c defold-docker-image/bob
no containers matched your search parameters!

they may have expired if your build hasn't recently finished.
```

- Do any of the check containers have a running /opt/resource/check process? If so, that may be hanging. What resource type is it?

N/A

- What is the uptime of your workers and ATC?

2 mins, 45 seconds for both

- Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?

Option 2 I guess?  I'm using the vagrant image from the website.

- Which IaaS?

I don't know.

- If you're on GCP, have you configured the MTU of your workers to be 1460 to match the VM? If not, it defaults to 1500, which would cause things to hang.

GCP == google cloud platform?  Then no, not on GCP...

- Can you reach the workers from your ATC? (curl http://<worker ip:port>/containers) You can collect the IP + port from fly workers -d.

Yes:
```
$ curl http://192.168.100.4:7777/containers
{"Handles":["6d1bfc41-4071-4699-5d15-b756c4d861c1"]}
```

**Comment by vito** [comment]
@ericroy Looks different from #750 - could you check your ATC logs? Maybe it's failing to create the container for the resource using the custom type. (Maybe the custom type is failing to check?)

**Comment by ericroy** [comment]
Indeed, you're right, something else is going on:

In the logs, I see this:
```json
{"timestamp":"1487706775.337479591","source":"atc","message":"atc.syncer.defold-docker-image:radar.scan.tick.check-image.init.failed-to-look-for-existing-container","log_level":2,"data":{"error":"invalid container identifier","pipeline-scoped-name":"defold-docker-image:resource-type:defold-bob-resource","session":"15.3.1.3.1.1"}}
{"timestamp":"1487706775.339088202","source":"atc","message":"atc.syncer.defold-docker-image:radar.scanner-failed","log_level":2,"data":{"error":"invalid container identifier","member":"defold-docker-image:resource-type:defold-bob-resource","session":"15.3"}}
```

And in my pipeline I have this:
```yaml
resource_types:
  - name: defold-bob-resource
    type: docker-image
    source:
      repository: eroy/defold-bob-resource

resources:
  - name: bob
    type: defold-bob-resource
    check_every: 6h
```

As a test, I changed the name of my custom resource from `defold-bob-resource` to `dbr`, but this had no effect - I get a similar error.  Any ideas?

**Comment by vito** [comment]
@ericroy Ah, looks different. Your resource needs a `source` configured. Try `source: {interval: 1h}` instead of `check_every: 6h`.

**Comment by ericroy** [comment]
That didn't seem to help, I still have this in the log:
```json
{"timestamp":"1487709643.636042595","source":"tsa","message":"tsa.connection.keepalive","log_level":1,"data":{"remote":"127.0.0.1:33456","session":"1","type":"keepalive"}}
{"timestamp":"1487709647.530714989","source":"atc","message":"atc.worker-collector-runner.lock-task.tick","log_level":1,"data":{"session":"19.2","task-name":"worker-collector"}}
{"timestamp":"1487709647.554697990","source":"atc","message":"atc.worker-collector-runner.lock-task.run-task","log_level":1,"data":{"session":"19.2","task-name":"worker-collector"}}
{"timestamp":"1487709647.567058086","source":"atc","message":"atc.syncer.defold-docker-image:radar.start","log_level":1,"data":{"session":"15.3"}}
{"timestamp":"1487709647.567564726","source":"atc","message":"atc.syncer.defold-docker-image:scheduler.start","log_level":1,"data":{"inverval":"10s","session":"15.4"}}
{"timestamp":"1487709647.687927008","source":"atc","message":"atc.syncer.defold-docker-image:radar.scan.tick.init.expiring-redundant-volume.expiring","log_level":1,"data":{"pipeline-scoped-name":"defold-docker-image:resource:dockerhub","session":"15.3.3.3.1.1","volume-handle":"cd318269-d0ef-4e89-794a-47e7c01b8254"}}
{"timestamp":"1487709647.713614941","source":"atc","message":"atc.syncer.defold-docker-image:radar.scan.tick.init.expiring-redundant-volume.expiring","log_level":1,"data":{"pipeline-scoped-name":"defold-docker-image:resource:dockerhub","session":"15.3.3.3.1.2","volume-handle":"7e1832b8-0d12-4ad6-5dc6-48f904446e50"}}
{"timestamp":"1487709647.731667995","source":"atc","message":"atc.syncer.defold-docker-image:radar.scan.tick.init.expiring-redundant-volume.expiring","log_level":1,"data":{"pipeline-scoped-name":"defold-docker-image:resource:dockerhub","session":"15.3.3.3.1.3","volume-handle":"3646c963-d96c-458b-6ee0-ffeeab66dc8f"}}
{"timestamp":"1487709647.753224373","source":"atc","message":"atc.syncer.defold-docker-image:radar.scan.tick.init.expiring-redundant-volume.expiring","log_level":1,"data":{"pipeline-scoped-name":"defold-docker-image:resource:dockerhub","session":"15.3.3.3.1.4","volume-handle":"15dea5e8-1dd0-4062-76d3-6f9949181cf5"}}
{"timestamp":"1487709647.772664070","source":"atc","message":"atc.syncer.defold-docker-image:radar.scan.tick.init.expiring-redundant-volume.expiring","log_level":1,"data":{"pipeline-scoped-name":"defold-docker-image:resource:dockerhub","session":"15.3.3.3.1.5","volume-handle":"da8d23b7-a994-400d-7528-c8bd43201dd8"}}
{"timestamp":"1487709647.786969185","source":"atc","message":"atc.syncer.defold-docker-image:radar.scan.tick.init.expiring-redundant-volume.expiring","log_level":1,"data":{"pipeline-scoped-name":"defold-docker-image:resource:dockerhub","session":"15.3.3.3.1.6","volume-handle":"6ed2e4da-26e6-44c2-6531-0d5cc570a78f"}}
{"timestamp":"1487709647.791584730","source":"atc","message":"atc.syncer.defold-docker-image:radar.scan.tick.init.expiring-redundant-volume.expiring","log_level":1,"data":{"pipeline-scoped-name":"defold-docker-image:resource:dockerhub","session":"15.3.3.3.1.7","volume-handle":"367ebbf9-e2c9-4e73-586c-e6505fcebf0c"}}
{"timestamp":"1487709647.793588638","source":"atc","message":"atc.syncer.defold-docker-image:radar.scan.tick.check-image.init.failed-to-look-for-existing-container","log_level":2,"data":{"error":"invalid container identifier","pipeline-scoped-name":"defold-docker-image:resource-type:dbr","session":"15.3.1.3.1.1"}}
{"timestamp":"1487709647.805517673","source":"atc","message":"atc.syncer.defold-docker-image:radar.scanner-failed","log_level":2,"data":{"error":"invalid container identifier","member":"defold-docker-image:resource-type:dbr","session":"15.3"}}
```

Also, I had omitted `source`, and used `check_every` based on the docs here:  https://concourse.ci/single-page.html#configuring-resources

Are those docs up to date with respect to 2.7?

**Comment by vito** [comment]
Yep, the docs are up-to-date.

I recall seeing that error before, don't remember what exactly triggers it. Maybe @mariash remembers? In any case, that should be fixed in 2.8 once it's out - we've completely redone how we handle the lifecycle of containers and volumes (per #629).

**Comment by ericroy** [comment]
Ok cool - I'm not really blocked at the moment, so I can probably just wait for 2.8 if it comes to that.

**Comment by troykinsella** [comment]
I'm seeing this issue with `2.7.0` binaries.

![concourse_2 7 0_frozen_job](https://cloud.githubusercontent.com/assets/1655907/23239525/a43dd770-f91c-11e6-92f6-09ba54c11d02.png)

- Are all of your workers present? Yes
- Is there a check container present for each of the inputs to your job? Sorry, had to downgrade before I checked this!
- What is the uptime of your workers and ATC? 1 hour
- Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)? Forwarded through the TSA
- Which IaaS? AWS
    - Can you reach the workers from your ATC? Ping, yes. Can't seem to connect as suggested. However, I don't see this stall issue with `2.6.0`.

[concourse_stack.tar.gz](https://github.com/concourse/concourse/files/795280/concourse_stack.tar.gz)

**Comment by troykinsella** [comment]
I noticed that when this bug occurs, `fly check-resource` on certain resources, but not all, that are "gotten" by the stalled job sit idle indefinitely, even though the UI reports that they're being `check`ed just fine. 

Is there any status update for this bug? It is starting to become a show stopper for my organization, sadly.

**Comment by freelock** [comment]
Kernel dmesg log here: https://concourseci.slack.com/files/freelock/F4GBU7MA8/-.sh ...

The worker lifecycle, ssh overflow, and ubuntu kernel oom issue have all been fixed and the system is feeling a lot more stable these days -- until we hit this one.

On this worker, it hums away for days at a time with a relatively light amount of use with no issue. As soon as we start stacking up jobs, it tends to kick in, and the system load spikes to around 200!?! (didn't even know that was possible...)

If I stop the concourse-worker, it takes a while to exit gracefully, but the load drops immediately as soon as I do.

This time I restarted the concourse worker (process only, not the entire host) and it seems to be doing fine. A few days ago when I restarted the worker, the load jumped back up to ~200 and stayed that way for an hour or two with no active jobs at all. Lots of btrfs processes. When I stopped it again, the ATC continued to show a bunch of active connections... after restarting both the worker and the ATC everything died back down and ran completely fine for 3 days until it just happened again now...

**Comment by gerhard** [comment]
We have come across the same:

```
+--------------------+-----------------------------+--------------------------------------------------+--------------+
| Name               | Release(s)                  | Stemcell(s)                                      | Cloud Config |
+--------------------+-----------------------------+--------------------------------------------------+--------------+
| concourse-rabbitmq | concourse/2.7.0             | bosh-vsphere-esxi-ubuntu-trusty-go_agent/3363.12 | latest       |
|                    | garden-runc/1.1.1           |                                                  |              |
+--------------------+-----------------------------+--------------------------------------------------+--------------+

+-------------------------------------------------+---------+----+---------+-------------+-----------------------------------------+--------------------------------------+--------------+--------+
| VM                                              | State   | AZ | VM Type | IPs         | CID                                     | Agent ID                             | Resurrection | Ignore |
+-------------------------------------------------+---------+----+---------+-------------+-----------------------------------------+--------------------------------------+--------------+--------+
| db/0 (0bb048a1-e5ad-4997-8fab-7f3ddc945521)     | running | z1 | 2c4gb   | 10.194.7.38 | vm-e318b0fb-b866-4f16-8601-00a7e2e9d9a7 | 304df294-6bd7-47fa-95e1-ca6dcf142566 | active       | false  |
| web/0 (e1d50b82-1614-4c0d-bbd3-801bd16bff48)    | running | z1 | 2c4gb   | 10.194.7.37 | vm-a13b1609-241b-4faa-8d38-992c5887f9fd | a36aa91e-ade2-447a-b9d3-08819cd92e89 | active       | false  |
| web/1 (bda6eda9-8859-411d-a659-9f00e831dfb3)    | running | z1 | 2c4gb   | 10.194.7.41 | vm-565b4831-e0cb-4466-b203-e28e7c5963f7 | 9990e5b5-af1b-4ed5-bf53-e6e75271bff5 | active       | false  |
| worker/0 (3435787d-d9d1-4fbc-a920-d9d5873c34ee) | running | z1 | 16c32gb | 10.194.7.39 | vm-d73a1ac0-f52e-499e-a874-b9afef9ec64e | b911b1b6-1be5-4336-a265-1911974305cb | active       | false  |
| worker/1 (12f47740-e9b4-4c04-a039-9fc374ac9640) | running | z1 | 16c32gb | 10.194.7.40 | vm-d6685d77-b17a-4e00-b6b1-faa34397420f | cf99037c-e464-4e97-8fb5-c96b0d091d6f | active       | false  |
| worker/2 (533c76e4-38fe-4917-80af-0458be2e01d7) | running | z1 | 16c32gb | 10.194.7.43 | vm-dca4e92d-a939-44d4-b2de-f5f479fa63de | e86a3ea8-feba-4d86-b69d-63ae4ffc2ca1 | active       | false  |
| worker/3 (0774f1af-b4f0-4d40-93e5-fb45cd6cc1c9) | running | z1 | 16c32gb | 10.194.7.42 | vm-8f287fad-edc5-498b-af5f-9c95c98163d0 | 04019bac-b135-4411-b4f1-7aae8aa7f998 | active       | false  |
+-------------------------------------------------+---------+----+---------+-------------+-----------------------------------------+--------------------------------------+--------------+--------+

0774f1af-b135-4411-b4f1-7aae8aa7f998  30  linux  none  none  running  10.194.7.42:7777  http://10.194.7.42:7788  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
12f47740-e464-4e97-8fb5-c96b0d091d6f  28  linux  none  none  running  10.194.7.40:7777  http://10.194.7.40:7788  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
3435787d-1be5-4336-a265-1911974305cb  23  linux  none  none  running  10.194.7.39:7777  http://10.194.7.39:7788  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
533c76e4-feba-4d86-b69d-63ae4ffc2ca1  28  linux  none  none  running  10.194.7.43:7777  http://10.194.7.43:7788  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
```

There are no stale workers or STDERR in ATC. [atc.stdout logs from web.0](https://gist.github.com/gerhard/2a7bed2d859fc6f74c901d802bb08d21). Starting and stopping the workers didn't fix the issue.

I have isolated the Concourse deployment in case you want to check it out. Switching to another Concourse deployment, on another IaaS, unblocked us.

cc @Osis @dumbbell @trevormarshall

**Comment by Osis** [comment]
@gerhard can you go through the steps provided in the original message so we can grab more context?

**Comment by cunnie** [comment]
* Are all of your workers present? (fly workers). All of the workers that our team can see & that our team uses are present and accounted for. Other teams have additional workers, but we can't see those:

```
fly -t cpi-tmp workers
name                                  containers  platform  tags                   team     state
063d37a8-8933-4450-6daa-84c4158e3787  0           linux     asia                   pivotal  running
773bfbb5-127c-4279-a0bd-901e4feee575  166         linux     none                   pivotal  running
9cd8bf62-51d2-4999-64ec-105a9b156496  2           linux     vsphere, vsphere-v5.1  pivotal  running
d7800d9e-5bd6-4393-5701-77764617013c  0           linux     vcloud-v5.5            pivotal  running
```

* Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)

**Yes:**

```
fly -t cpi-tmp i -c bosh-vsphere-cpi/bosh-cpi-artifacts
bash-4.3# ps ax
PID   USER     COMMAND
    1 root     {runc:[2:INIT]} /proc/self/exe init
  142 root     bash
  148 root     ps ax
bash-4.3#
```
* Do any of the check containers have a running /opt/resource/check process? If so, that may be hanging. What resource type is it?

**No**. See above.

* What is the uptime of your workers and ATC?

```
for VM in {concourse_cpi/{0,1,2},xxx_worker,pivotal_worker,yyy_worker}; do bosh ssh -d concourse-cpi --gw-user=vcap   --gw-host=10.0.0.6   --gw-private-key=$PWD/vcap.pem $VM uptime; done | grep "load average"
concourse_cpi/5d0a3dfb-48d9-4746-8df6-f77f344d3fc6: stdout |  18:41:58 up 5 days,  4:43,  1 user,  load average: 0.09, 0.10, 0.16
concourse_cpi/211dd45d-9fac-4cfe-b0e9-265c6dba6e91: stdout |  18:42:00 up 18 days, 18:46,  1 user,  load average: 0.12, 1.39, 2.05
concourse_cpi/30fec437-791d-4a07-8d58-e4fad07aeac9: stdout |  18:42:01 up 4 days, 49 min,  1 user,  load average: 0.04, 0.12, 0.09
xxx_worker/4152844b-27ca-4ffc-94ba-73564c1bc7c7: stdout |  18:42:02 up 24 days, 19:27,  1 user,  load average: 2.90, 2.99, 2.99
pivotal_worker/773bfbb5-db69-430a-8de7-8be952111b61: stdout |  18:42:03 up 24 days, 19:37,  1 user,  load average: 0.60, 0.92, 1.14
yyy_worker/c1cfe0bb-f0d2-4ab1-9f04-639a8d1144cf: stdout |  18:42:04 up 24 days, 19:32,  1 user,  load average: 0.10, 0.03, 0.01
```

* Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?

We believe _pivotal\_worker_ is registered directly, and the remaining workers (_xxx_, _yyy_) are `bosh-init` deployed external workers forwarded through TSA.

Which IaaS?

**GCP**

If you're on GCP, have you configured the MTU of your workers to be 1460 to match the VM? If not, it defaults to 1500, which would cause things to hang.

**Oops**, no we have not configured the MTU of our workers. Some our workers are on different IaaSes (i.e. vSphere, vCloud), so we're gonna skip those, but we'll redeploy our GCP worker with the smaller MTU.

* Can you reach the workers from your ATC? (curl http://<worker ip:port>/containers) You can collect the IP + port from fly workers -d.
In addition to that, the most valuable information will be stack dumps of both the ATC and the TSA.

```
for WORKER in 5d0a3dfb-48d9-4746-8df6-f77f344d3fc6.concourse-cpi.concourse.concourse-cpi.bosh:36557 30fec437-791d-4a07-8d58-e4fad07aeac9.concourse-cpi.concourse.concourse-cpi.bosh:35510 5d0a3dfb-48d9-4746-8df6-f77f344d3fc6.concourse-cpi.concourse.concourse-cpi.bosh:44896 5d0a3dfb-48d9-4746-8df6-f77f344d3fc6.concourse-cpi.concourse.concourse-cpi.bosh:38813; do curl $WORKER/containers; done
{"Handles":[]}
{"Handles":["024b63fd-a7b9-4278-5937-cb0046e08c70","02579075-143f-4ff5-7db1-e2f9fcb3a2be","02fc05ad-8595-495a-7a67-0cd5d87bf76e","04a12683-49fc-4538-4697-7336ca10312c","07c01710-fd1e-4c07-4f40-6011dc62de54","0c131f99-c8fb-4eb7-5cdf-4144b7998b84","0cd49c20-5029-438e-5a5f-08a6c95eb6bc","0d928f9c-a0f9-4e09-7fae-1deb3742133b","10f5f17e-1fc8-4f3e-507c-3d180999352a","115252f3-c756-416c-4ae8-b10d324f6225","14c724b8-7f90-43be-6c61-e96ea72c7c77","15bdae0b-0483-4d44-47e1-f45a51b1952d","1b8c2b76-d249-4394-4231-eb3bec782d97","1beb2893-769e-4c3f-5920-85afb2797213","1d0ea82c-7207-4c8b-780d-914bc0cef12b","216fc4aa-52b3-4f51-4d6c-12bf8f574263","21ad4cd4-c2fe-4995-4ca3-72122dba29ee","240da70a-93d2-46d9-7461-797ed3f4143b","294712e3-0c40-46c5-50e5-149ff8432528","2ae44092-b7e5-4495-6f4a-e2e9a50865e1","2d827847-89fb-4582-7237-ac40725c4ac7","2dc18ba5-8571-4b06-4aca-95c0e318d323","2e23582f-da8e-4ad8-46d3-52816f8e1069","2e9cafdc-f6a2-4d68-45d2-1ca505a1d87e","2f904fc1-64bc-4ffb-656a-af0bfef48100","2fb8d27f-614e-4e69-4c7e-89dc5d31ee43","3b79248f-6ec6-4bb3-4a90-ee04e065a13d","3bdff5d3-65b6-4771-701e-0cdc11a9c478","3c9937d2-57d9-4e73-64fe-d46e9d9a3682","3ca323ef-ce62-4f63-43a8-19b9cebeb2a3","3e3b9214-aa8c-415f-6dc1-d5d000ae6951","409288f8-f763-4698-5775-037af9830c22","4096879e-861d-4c32-7e81-69919adbfae2","41aea8f3-1d4a-47a6-450c-a51f10d7c484","4847534f-91dd-4f6e-6c0a-243db8d91b2f","490f0075-2e4c-4b0c-4d91-2b196572e2ff","4e8691ff-f3cd-470c-6149-b226a8a56005","4e9dca98-ce60-406c-6444-1f3525a246b5","54ca04ed-d819-49ec-76d7-35fd51e9962a","57f14ac5-49fc-4eb7-4fe1-29dc5ccad9fb","5837b37d-9342-4d18-6eb9-167f2a3edba6","5896f3ac-42aa-41a8-4352-a085707754ba","5933672f-b5ee-49b4-578f-99c8213845a2","5a264268-0cc2-4018-75c1-c060058f52f8","5cab7316-9c06-42d0-7829-252b5f2ddef0","5f8238f4-fb3e-443d-613a-7111a5f3e3de","602b23f4-a24c-499b-6ec4-98c5ea66a19c","619e9621-b604-4ffd-7f43-9bd1b73c8b9f","62135911-a3ed-4eeb-6772-83f6d4934124","68e889e7-4d79-4d10-74c4-6e8e70a787ed","6ab3a35f-a67d-4ba9-6c02-49e82cfb0b97","6b8db127-4062-41c4-60b9-135e08ca65d5","6d624008-44bf-42d9-5670-e85775123947","701ebbeb-8355-4039-6d1f-ad53189be454","7b86e90f-2b35-4d50-6a57-ef37a0e9878b","7bf969c2-e5a4-4aef-4943-6d70f0135b47","7c848aa9-f635-4ef4-5b10-e28ac7761308","80f95661-3930-4c66-7320-895fafda9856","835503a5-906e-4f12-79c2-e09fbc604198","8408b896-f378-4364-6a20-6ab2be176987","86a25903-23ec-44ca-67fb-258488288670","8ab25af1-35e0-49e4-79bb-6176ce0cd726","8ae5014d-a2fd-4eb3-7323-078f9cdbd53e","8b630734-c515-4bad-6c88-760efc48b9c1","8b9594b2-d457-4992-7964-9017e52cc7cd","8de6ebce-edcd-482d-6a9d-bf8b3e80e509","8e676eba-4657-4ec9-7cb7-2b1a5c46a48b","907040d7-4773-4cd6-7b88-a01fd6ddc964","92c1ddb9-d85a-4e01-55ae-4e70fb31fdad","9502f69a-0104-4afd-5702-8b1d13e117a0","97ff88b7-5df6-451e-564e-f28e162294da","996bbbd3-7c40-41f6-4219-4cfea84dd71c","9e138942-b14c-4282-52f0-976735a46632","9fb9a6d3-35fa-4848-608e-57bf8dd95c96","a01c463f-6e7a-45cc-5a23-e78bcaeffc0a","a0b30af2-6657-478b-7dd9-10e6beeec9ae","a50682a6-102a-4a2e-7a4c-811208981c86","a5b0ce8a-a971-49e9-515c-755e3e5b94a1","a781b8f1-9cb9-4920-5347-c0d6c794327b","a8482371-b9c5-40bb-72bf-b6df27fab493","a9d3256a-9c3d-4c92-7081-284de4437741","b1405753-5193-4266-6408-8583a60659a6","b208a7d1-215d-4096-564e-1a880c8f8f6a","b40c7d15-4538-4799-7d89-6f1f56b4bb66","b452e0d5-5ac2-4643-5b22-bee79a8a7451","b7dd8d4e-bd77-438d-6eb7-e14a939e5999","b89104d9-5cf3-4450-4694-58f0a49375ba","bacfd67d-d08b-4283-463b-76957e5c5cd6","bbb83688-167d-4e91-7baf-c76afff7970d","be8d3a34-bb6f-47de-753e-25d609a56418","c0b1e120-713a-495a-5ce8-d8e14d48a0df","c1376ca2-14c6-4fbe-4aaf-26504680d5b7","c60407af-ee77-41eb-543e-89247be6b3d8","c9339838-3258-4c55-708d-fb27cccf5c40","cc113330-e22d-4cca-7ad9-7acce3dc492c","cc3e19ce-2b00-463b-7703-3966e8ce0776","ccc4bc10-ae5a-4981-477e-5bb87739b8ba","d5c8f0e5-beba-4081-6fab-5d343edf16dd","df3ff71e-cd90-43cc-7827-036379202448","e1ed81b9-a1b9-4325-69a7-c89882cf3860","e5e6a9bd-18d0-4324-4026-ec29ed42e354","efb28e29-a4eb-44d0-4a14-b4412ad95a80","f0b40b3e-6f85-4cc2-5f3a-2dd0257605ed","f306d199-c6c2-4c35-6ae2-67e8abc60322","f36a3a08-47d3-4f7f-64d8-97f6797e5eb0","f3cb7bba-254d-4de4-73bf-ac86ec67392c","fdba771c-636f-40dc-6bd0-48798ef77a98","ff2341c1-29d3-4448-475f-080354e1140c"]}
{"Handles":["0c2153a0-ce5e-4955-76f2-dfd9cf5ee21f","1945dc45-c23f-4de8-429a-f2be9a35a92a"]}
{"Handles":[]}
```

* You can collect a stack dump from the ATC by running:

curl http://127.0.0.1:8079/debug/pprof/goroutine?debug=2

Stack dump is visible here <https://nono.io/bosh-cpi-atc-pprof.txt>

...and from the TSA by sending SIGQUIT and collecting the output from stderr. Note that if you're running the binaries, the above curl command will include the TSA's stack, so don't worry about getting it separately. Also note that SIGQUIT will kill the TSA process, so you'll need to bring it back after. (While you're at it, let us know if that fixed it. :P)

Stack dump is visible here <https://nono.io/tsa.stderr.log>

**Comment by quantonganh** [comment]
>If you're seeing this problem, please include a screenshot of the build preparation section of the build view.

```
preparing build
checking pipeline is not paused
checking job is not paused
discovering any new versions of code
discovering any new versions of therecipe-qt-windows
waiting for a suitable set of input versions
code - no versions available
therecipe-qt-windows - no versions available
checking max-in-flight is not reached
```
> Are all of your workers present? (fly workers)

```
name               containers  platform  tags  team  state    garden address  baggageclaim url       resource types
sad-ibex-worker-0  0           linux     none  none  running  10.0.3.5:35268  http://10.0.3.5:45868  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
sad-ibex-worker-1  0           linux     none  none  running  10.0.3.5:46744  http://10.0.3.5:39357  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
```

but I'm wondering why `containers` is zero?

> Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)

No, both workers returned:

```
no containers matched your search parameters!

they may have expired if your build hasn't recently finished.
```

>What is the uptime of your workers and ATC?

```
 04:22:34 up 2 days, 15:29,  0 users,  load average: 0.01, 0.08, 0.08
```

> Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?

It is installed via `helm`: https://github.com/kubernetes/charts/tree/master/stable/concourse
with some custom values:

```
imageTag: "2.7.0"

web:
  service:
    type: LoadBalancer

persistence:
  worker:
    size: 100Gi
```

and I guess workers are forwarded through the TSA.

>Which IaaS?

Google Compute Engine.

>If you're on GCP, have you configured the MTU of your workers to be 1460 to match the VM? If not, it defaults to 1500, which would cause things to hang.

Why I saw this [here](https://cloud.google.com/compute/docs/troubleshooting):

> The Google Compute Engine network has a maximum transmission unit (MTU) of 1460 bytes. The operating system images provided by Compute Engine are configured with this MTU, so no action is required if you're using one of those images. For custom images, set the MTU to 1460 to avoid increased latency and packet overhead caused by fragmentation.

So, where should I set the MTU? in GCE or in the custom image?

> Can you reach the workers from your ATC? (curl http://<worker ip:port>/containers) You can collect the IP + port from fly workers -d.

```
name               containers  platform  tags  team  state    garden address  baggageclaim url       resource types
sad-ibex-worker-0  0           linux     none  none  running  10.0.3.5:35268  http://10.0.3.5:45868  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
sad-ibex-worker-1  0           linux     none  none  running  10.0.3.5:46744  http://10.0.3.5:39357  archive, bosh-deployment, bosh-io-release, bosh-io-stemcell, cf, docker-image, git, github-release, hg, pool, s3, semver, time, tracker, vagrant-cloud
```

```
curl http://10.0.3.5:35268/containers
{"Handles":[]}

curl http://10.0.3.5:46744/containers
{"Handles":[]}
```
>In addition to that, the most valuable information will be stack dumps of both the ATC and the TSA.


[atc.txt](https://github.com/concourse/concourse/files/871493/atc.txt)

**Comment by topr** [comment]
I'm facing the similar issue as @ericroy
Just a Vagrant/VirtualBox image provided. Nothing fancy. Very simple pipeline and a custom resource type (`npm-cache-resource`). There is no even a container in state check just UI shows:
>discovering any new versions of webapp-npm-cache
>waiting for a suitable set of input versions
It goes forever as I left over 72 hours and i hasn't gone further. The same pipeline but with other resource (a built in git resource) works fine (both use the same source properties as npm-cache-resources fetch repo from git anyway).

It's my first attempt of using Concourse so not sure about the debug I could do yet.
Version 2.7.3.

**Comment by adi-mishra** [comment]
Hey, we were seeing this same issue over and over again, but it was fixed after we recreated a stalled worker. Hope this helps.

**Comment by JohannesRudolph** [comment]
> If you're seeing this problem, please include a screenshot of the build preparation section of the build view.

![image](https://cloud.githubusercontent.com/assets/130103/25854290/ca220a26-34cf-11e7-8d12-d8afa165154a.png)

> Are all of your workers present? (fly workers)
No, but that's expected since we run concourse using docker-compose and sometimes we recreate the workers (e.g. when upgrading concourse versions): 
```
PS C:\dev\mc\meshfed> fly -t ci workers
name          containers  platform  tags  team  state
6b6a8d60c5f7  20          linux     none  none  running


the following workers have not checked in recently:

name          containers  platform  tags  team  state
79a1aaa96e7a  1           linux     none  none  stalled
9237f14faf4a  8           linux     none  none  stalled
b2e592651535  0           linux     none  none  stalled ```



> Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)
Yes
> Do any of the check containers have a running /opt/resource/check process? If so, that may be hanging. What resource type is it?
we're using the vanilla git resource. Depending on the time I check with hijack an `/opt/resource/check` may run or may not. In any case it doesn't appear stalled

> What is the uptime of your workers and ATC?
Very few minutes, just `docker-compose up -d`'d. 

> Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?
Directly I supppose. 

> Which IaaS?
We're running on meshcloud.io OpenStack, where the network (and garden) MTU is both 1500. 

> In addition to that, the most valuable information will be stack dumps of both the ATC and the TSA.
Here you go: https://gist.github.com/JohannesRudolph/eab735e236303adbfdaf31c8583d7205

The build didn't get unstuck after killing the ATC + TSA.

**Comment by dmlemos** [comment]
Today we had a number of builds with this problem.

> Are all of your workers present? (fly workers)

Yes

> Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)

Empty

> What is the uptime of your workers and ATC?

I had restarted ATC recently due to other issue.

`atc: up  2:12`
`worker: up 1 day`

> Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?

BOSH

> Which IaaS?

AWS

> Can you reach the workers from your ATC? (curl http://<worker ip:port>/containers) You can collect the IP + port from fly workers -d.

yes

> In addition to that, the most valuable information will be stack dumps of both the ATC and the TSA.

https://gist.github.com/dmlemos/287a1bcf65f2fd733065abd204e051b9
https://gist.github.com/dmlemos/bd6c9925977518eef369a79847ba259a

---
Small part of the baggageclaim logs from the worker:
```
{"timestamp":"1494849125.302422523","source":"baggageclaim","message":"baggageclaim.driver.run-command.failed","log_level":2,"data":{"args":["qgroup","show","-F","--raw","/var/vcap/data/baggageclaim/volumes/live/efcabf18-7d55-45a6-5ea0-22082551fa41/volume"],"command":"/var/vcap/packages/btrfs_tools/sbin/btrfs","error":"exit status 1","session":"1.338713","stderr":"ERROR: can't perform the search - No such file or directory\nERROR: can't list qgroups: No such file or directory\n","stdout":""}}
{"timestamp":"1494849125.302480459","source":"baggageclaim","message":"baggageclaim.repository.get-volume.failed-to-get-volume-stats","log_level":2,"data":{"error":"exit status 1","session":"3.2909157","volume":"efcabf18-7d55-45a6-5ea0-22082551fa41"}}
{"timestamp":"1494849191.070497513","source":"baggageclaim","message":"baggageclaim.tick.reaping","log_level":1,"data":{"handle":"9062b9ec-1eb1-40b2-4f7c-c479b9b05776","session":"15944","ttl":300}}
```

Cancelling the build did nothing. Killing the TSA nothing as well.
Only recreating the worker fixed it.

**Comment by jgeorge300** [comment]
<img width="520" alt="screen shot 2017-06-28 at 11 33 56 am" src="https://user-images.githubusercontent.com/513514/27646023-a081a3be-5bf5-11e7-860b-a19e9745d621.png">

* Are all of your workers present? (fly workers)
Yes

* Is there a check container present for each of the inputs to your job? (fly hijack -c pipeline-name/resource-name)
1 of 2 resources was missing a check container. 

* Do any of the check containers have a running /opt/resource/check process? If so, that may be hanging. What resource type is it?
The existing container is a docker-image

* What is the uptime of your workers and ATC?
Our concourse instances are stopped / started daily. 

Are your workers registered directly (BOSH default) or forwarded through the TSA (binary default/external workers registering with BOSH deployment)?
TSA

* Which IaaS?
AWS

Started occurring when we upgraded to 3.1.1 (seeing it on 3.2.1 as well) Deleting the pipeline and re-adding fixes the issue. Our concourse instances are turned off in the evening and re-started in the morning. Starting a new job immediately goes into a hang state. Our images are created similar to the concourse ami's in aws.

**Comment by kevinzurek** [comment]
I ran into this same issue today (stuck on "discovering new versions") on a bosh deployed instance of Concourse 2.7.3 and noticed 1 of the 3 workers had a very high load. I was able to resolve the problem by manually sshing into the offending worker and restarting it.

Here's a snippet of the container load before restarting:
```
name                                  containers  platform  tags  team  state
98b9dcfa-e345-4317-bcbf-ba5cbfbb622a  15          linux     none  none  running
a1435e37-a2e3-40e0-98fa-2c257f7666d2  74          linux     none  none  running
afa869ea-e91c-4054-918d-3e05b6034053  12          linux     none  none  running
```

And an image of the worker load:
![concourse_workers_hanging](https://user-images.githubusercontent.com/5403109/27965405-b0b48b6e-6309-11e7-9efd-70aeb0e36b2a.png)

After the worker came back up it seems like the container load was properly distributed again:
```
name                                  containers  platform  tags  team  state
98b9dcfa-e345-4317-bcbf-ba5cbfbb622a  72          linux     none  none  running
a1435e37-c887-447e-9862-49ca3a5bd217  48          linux     none  none  running
afa869ea-e91c-4054-918d-3e05b6034053  64          linux     none  none  running
```

**Comment by Murryy79** [comment]
I am having similar issue. stuck at discovering resource .  resource  type is docket image. repository cloud-build-org/bosh-creds-resource exists and and accessible in our local github . 
resource_types:
- name: bosh-creds
  type: docker-image
  source:
    repository: cloud-build-org/bosh-creds-resource
    tag: latest

My concourse is v2.7.4. 

I delete pipeline a few times and recreate but no luck

![image](https://user-images.githubusercontent.com/19574664/28321093-eaaee666-6b97-11e7-9f56-261400d6972f.png)

**Comment by G2Jose** [comment]
Same issue here
![image](https://user-images.githubusercontent.com/6994514/28956314-0cd85ebe-78ba-11e7-92c8-450ef050ca9d.png)

**Comment by Murryy79** [comment]
In my case I had incorrect docker image repo configured for the inputs that were indefinitely waiting for discovering. Corrected that and was all good.

**Comment by vito** [comment]
I'm going to close this as we fixed the underlying root cause for the many reports we received at the time. All the recent reports appear to be a misconfiguration or some other problem with Concourse itself working as expected (build is pending and it tells you why, e.g. there are no versions, which is up to you to solve).

**Comment by JohannesRudolph** [comment]
Respectfully disagree with the assertion that all issues are resolved. I
produced very clear reproduction instructions above for issues we have with
concourse when re-deploying workers. These problems lead to exactly the
symptoms discussed in this issue. Shall we track them as a separate issue
instead maybe? I agree this issue has become somewhat of a dumping ground
(but with a little work maybe makes for a good FAQ what to look for if
you're builds are stuck pending)

**Comment by vito** [comment]
@JohannesRudolph Definitely didn't mean to imply that all issues are resolved. My point was more that this issue has become derailed with many different cases, some of which are user error, and some of which are other things to fix/improve in Concourse. There was a very specific fix we made a long while back that fixed this for the largest portion of affected users. Anything else should be a separate issue, if not only for a clearer discussion but also because the debugging steps may end up being different.

**Comment by youngm** [comment]
Any luck @jgeorge300?  Your hang situation seems to match ours.  Deleting the pipeline and re-adding seems to fix it.

**Comment by nsmith-pivotal** [comment]
If this is still happening to you in 2019, we were able to find an error in the `stderr.log` of our main web worker that mentioned a failed TLS handshake.
```
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2884 +0x2f4
2019/05/24 13:55:18 http: TLS handshake error from 108.178.16.154:35882: tls: first record does not look like a TLS handshake
```
We were able to solve our issue by restarting the web process on our main web worker. We are using v `5.0.0`. This way you don't have to delete and re-add pipelines. We do not know what the root cause is.

**Comment by vito** [comment]
I'm gonna lock this thread since it's hard to keep track of conversations in closed issues. :slightly_smiling_face: 

If anyone is still running into this problem, please [take a stack dump](https://github.com/concourse/concourse/wiki/Taking-a-Stack-Dump) and attach it to a new issue to help with debugging.

The latest instance of this is #3928 which is prioritized highly in the Runtime backlog. The issue includes a stack dump if anyone wants to verify that their pending builds are stuck for the same reason.
